### PR TITLE
Add support for mouse scroll

### PR DESCRIPTION
**Pull request by fernyb**
Add support for mouse scroll as I couldn't get the cy.trigger("wheel") to work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -293,6 +294,30 @@ Options:
 - `Optional` **position**: "topLeft" | "top" | "topRight" | "left"  | "center" | "right" | "bottomLeft" | "bottom" | "bottomRight"
 - `Optional` **scrollBehavior**: "center" | "top" | "bottom" | "nearest" | false
 
+## cy.realMouseWheel
+
+Fires native system mouse up event.
```
Comment (dmtrKovalenko) [inline README.md:299]:
```suggestion
Fires native system mouse wheel event, scrolls the page. 
```

File: README.md [hunk 2]
```diff
@@ -293,6 +294,30 @@ Options:
...
+cy.get("div").realMouseWheel(options);
+```
+
+Example:
+
+```js
+cy.get("div").realMouseUp({ position: "center", deltaY: 500 }) // scroll on the center of div 500 pixel vertically
+cy.get("div").realMouseUp({ position: "center", deltaY: -500 }) // scroll on the center of div -500 pixel vertically
+cy.get("div").realMouseUp({ position: "center", deltaX: -500 }) // scroll on the center of div -500 pixel horizontally
+cy.get("div").realMouseUp({ position: "center", deltaX: 500 }) // scroll on the center of div 500 pixel horizontally
```
Comment (dmtrKovalenko) [inline README.md:311]: Please fix the examples.

File: cypress/integration/wheel.spec.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+describe("cy.realMouseWheel", () => {
+
+  it("scroll to find a day", () => {
+    cy.visit("https://anvil.servicetitan.com/components/date-picker/")
```
Comment (dmtrKovalenko) [inline cypress/integration/wheel.spec.ts:4]: Please move the code to our fixtures (preferred) or at least to code sandbox so we will own it.

File: cypress/integration/wheel.spec.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    const input = () => cy.get(".Card.CodeDemo").eq(1).find("input");
```
Comment (dmtrKovalenko) [inline cypress/integration/wheel.spec.ts:6]: Why this should be a function?

File: cypress/integration/wheel.spec.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    input().click();
+
```
Comment (dmtrKovalenko) [inline cypress/integration/wheel.spec.ts:8]:
```suggestion
```

File: cypress/integration/wheel.spec.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    const findDate = (dayStr: string): void => {
+      let found = false;
+      const scrollToMonthYear = (): void => {
+        const content = () => cy.get(".k-widget.k-calendar.k-calendar-infinite");
+        content()
+        .realMouseWheel({ scrollBehavior: false, position: "center", deltaY: -500 })
+
+        cy.wait(3);
```
Comment (dmtrKovalenko) [inline cypress/integration/wheel.spec.ts:17]: Why do we need an explicit wait?

File: cypress/integration/wheel.spec.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            scrollToMonthYear();
+          }
+        });
+      };
+      scrollToMonthYear();
+    };
+
+    findDate("June 18, 2019");
+    input().should("have.value", "06/18/2019")
+
```
Comment (dmtrKovalenko) [inline cypress/integration/wheel.spec.ts:38]:
```suggestion
```

File: src/commands/mouseWheel.ts [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    }),
+  });
+
+  await fireCdpCommand("Input.dispatchMouseEvent", {
+    type: "mouseWheel",
+    x,
+    y,
+    deltaX,
+    deltaY,
+    modifier: 0,
```
Comment (dmtrKovalenko) [inline src/commands/mouseWheel.ts:61]: Should we support modifiers as well? I don't know wheel mouse with alt for example

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]

**Comment by dmtrKovalenko** [comment]
Thank you for this addition. Left a couple of comments

**Comment by dmtrKovalenko** [comment]
I see the 1 failling test, any ideas what can be the reason?

**Comment by fernyb** [comment]
The build says the failure is in "realTouch".

touches using provided 0 for one of the axis
https://829-315086372-gh.circle-artifacts.com/1/cypress/screenshots/touch.spec.ts/cy.realTouch%20--%20touches%20using%20provided%200%20for%20one%20of%20the%20axis%20%28failed%29.png

touches with a custom radius 
https://829-315086372-gh.circle-artifacts.com/1/cypress/screenshots/touch.spec.ts/cy.realTouch%20--%20touches%20with%20a%20custom%20radius%20%28failed%29.png

File: src/commands/mouseWheel.ts (hunk 8 above)
Comment (fernyb) [reply src/commands/mouseWheel.ts:61]: I don't see a need to support modifier. I don't recall a time I've ever used them.

File: cypress/integration/wheel.spec.ts (hunk 3 above)
Comment (fernyb) [reply cypress/integration/wheel.spec.ts]: moved to use a fixture file.

**Comment by fernyb** [comment]
This is what I found out. When test runs and scale is not 100% then test fails in certain cases. When the scale is 100% the all tests pass. So I've updated tests to set viewport small enough to get 100% scale.

**Comment by dmtrKovalenko** [comment]
What do you mean by scale? When window is zoomed? We have a guard for that and correctly handling this in `getCypressElementCoorsinate`

**Comment by fernyb** [comment]
> What do you mean by scale? When window is zoomed? We have a guard for that and correctly handling this in `getCypressElementCoorsinate`

image
This one fails because the scale is not 100%. In the screenshot it says 86%.

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]
It looks like your tests are basically failling both on CI and locally. 

Please resolve these issues and make sure CI is green 
image

**Comment by muratkeremozcan** [comment]
Just use this test

https://github.com/muratkeremozcan/multi-stage-caching/blob/master/cypress/component/hooks/usehooks.com/useLockBodyScroll/use-lock-body-scroll.comp-spec.jsx

**Pull request closed without merging**